File: Python/443.string_compression.py
from typing import List
class Solution:
    def compress(self, chars: List[str]) -> int:
        previous = ''
        count = 1
        current = -1
        for char in chars:
            if char != previous:
                if count > 1:
                    count = str(count)
                    for i in range(len(count)):
                        current += 1
                        chars[current] = count[i]
                current += 1
                chars[current] = char
                count = 1
            else:
                count += 1
            previous = char
        if count > 1:
            count = str(count)
            for i in range(len(count)):
                current += 1
                chars[current] = count[i]
        return current + 1

File: C++/Two Pointers/443. String Compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        char prev = ' ';
        int count = 1;
        int index = 0;
        for (char c : chars) {
            if (c != prev) {
                string group_length = to_string(count);
                if (prev != ' ') {
                    chars[index++] = prev;
                    if (group_length != "1") {
                        for (int i = 0; i < group_length.size(); i++) {
                            chars[index++] = group_length[i];
                        }
                    }
                }
                count = 1;
            }
            else {
                count++;
            }
            prev = c;
        }
        chars[index++] = prev;
        if (count > 1) {
            string group_length = to_string(count);
            for (int i = 0; i < group_length.size(); i++) {
                chars[index++] = group_length[i];
            }
        }
        return index;
    }
};
